### PR TITLE
Fea: Manual & Karma banning

### PR DESCRIPTION
**Pull request by mzegar**
You are not able to pass in `long` into a `Convar`, so I convert it from a string to a long.

### CODE REVIEW

**Comment by mzegar** [comment]
@kolexxx We need to test this before merging, couldn't do it on my local machine.

File: code/Game.Moderation.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+			break;
+		}
+	}
+
+	[ConCmd.Admin( Name = "ttt_removeban_steamid", Help = "Remove the ban from a user" )]
+	public static void RemoveBanWithSteamId( long steamId )
+	{
+		var bansRemoved = _bannedClients.RemoveAll( ( bannedClient ) => bannedClient.SteamId == steamId );
+		if ( bansRemoved > 0 )
+			FileSystem.Data.WriteJson( BanFilePath, _bannedClients );
```
Comment (mzegar) [inline code/Game.Moderation.cs:61]: I'll change this to only remove one, there shouldn't be multiple bans here.

File: code/Game.Moderation.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+			BanPlayer( client, reason, true );
+			break;
+		}
+	}
+
+	[ConCmd.Admin( Name = "ttt_ban_steamid", Help = "Ban the user with the steamid" )]
+	public static void BanPlayerWithSteamID( long steamId, string reason = "" )
+	{
+		foreach ( var client in Client.All )
```
Comment (mzegar) [inline code/Game.Moderation.cs:46]: I should give feedback on all these commands on whether or not the player was found.

File: code/Game.Moderation.cs [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+			if ( client.PlayerId != steamId )
+				continue;
+
+			client.Kick();
+			return;
+		}
+
+		Log.Warning( $"Unable to find player with steam id {rawSteamId}" );
+	}
+}
```
Comment (kolexxx) [inline code/Game.Moderation.cs:126]: Any reason why we need these kick commands? S&box already has this built-in.

File: code/Karma.cs [hunk 4]
```diff
@@ -216,7 +216,7 @@ private static void OnRoundEnd( Team winningTeam, WinType winType )
 			Rebase( player );
 
 			if ( Enabled && CheckAutoKick( player ) )
-				client.Kick();
+				Game.BanPlayer( player.Client, $"Karma did not meet minimum threshold! {MinValue}", false );
```
Comment (kolexxx) [inline code/Karma.cs:219]: I think we should only ban players for low karma for one game of ttt until it changes maps.

File: code/Game.Moderation.cs (hunk 3 above)
Comment (mzegar) [reply code/Game.Moderation.cs:126]: I don't think you can kick with `steamid`, you should be able to kick by name. I thought maybe we should just include it anyways.

File: code/Karma.cs (hunk 4 above)
Comment (mzegar) [reply code/Karma.cs:219]: Yeah that was my intent, but I just realize that it doesn't work that way. I'll fix it when I add the timed bans.

File: code/Game.Moderation.cs (hunk 3 above)
Comment (kolexxx) [reply code/Game.Moderation.cs:126]: > I don't think you can kick with `steamid`, you should be able to kick by name. I thought maybe we should just include it anyways.

Nobody is gonna be kicking people by their steamid

Comment (mzegar) [reply code/Game.Moderation.cs:126]: You have to be able to kick via steamid since duplicate names can occur.

Comment (kolexxx) [reply code/Game.Moderation.cs:126]: That is a facepunch problem. We should remove these commands.

**Pull request merged**